test/jdk/java/security/PEM/PEMCerts.java

```diff
@@ -0,0 +1,338 @@
+        8gOYV33zkPhziWJt4uFMFIi7N2DLEk5UVZv1KTLZlfPl55DRs7j/Sb4vKHpB17AO
+        meVknxVvifDVY0TIz57t28Accsk6ClBCxNPluPU/8YLGAZJYsdDXjGcndQ13s5G7
+        -----END CERTIFICATE-----
+        """;
+
+    private static final String ecCert = """
+        -----BEGIN CERTIFICATE-----
+        MIIBFzCBvgIJAOGVk/ky59ojMAoGCCqGSM49BAMCMBMxETAPBgNVBAMMCFBFTSB0
+        ZXN0MCAXDTI0MDEwOTIzMzEwNloYDzIwNTEwNTI2MjMzMTA2WjATMREwDwYDVQQD
+        DAhQRU0gdGVzdDBZMBMGByqGSM49AgEGCCqGSM49AwEHA0IABGYI0jD7JZzw4RYD
+        y9DCfaYNz0CHrpr9gJU5NXe6czvuNBdAOl/lJGQ1pqpEQSQaMDII68obvQyQQyFY
+        lU3G9QAwCgYIKoZIzj0EAwIDSAAwRQIgMwYld7aBzkcRt9mn27YOed5+n0xN1y8Q
+        VEcFjLI/tBYCIQDU3szDZ/PK2mUZwtgQxLqHdh+f1JY0UwQS6M8QUvoDHw==
+        -----END CERTIFICATE-----
+        """;
+
+    // EC cert with explicit parameters -- Not currently supported by SunEC
+    private static final String ecCertEX = """
+        -----BEGIN CERTIFICATE-----
+        MIICrDCCAjMCCQDKAlI7uc1CVDAKBggqhkjOPQQDAjATMREwDwYDVQQDDAhQRU0g
+        dGVzdDAgFw0yNDAxMDkyMzIxNTlaGA8yMDUxMDUyNjIzMjE1OVowEzERMA8GA1UE
+        AwwIUEVNIHRlc3QwggHMMIIBZAYHKoZIzj0CATCCAVcCAQEwPAYHKoZIzj0BAQIx
+        AP/////////////////////////////////////////+/////wAAAAAAAAAA////
+        /zB7BDD//////////////////////////////////////////v////8AAAAAAAAA
+        AP////wEMLMxL6fiPufkmI4Fa+P4LRkYHZxu/oFBEgMUCI9QE4daxlY5jYou0Z0q
+        hcjt0+wq7wMVAKM1kmqjGaJ6HQCJamdzpIJ6zaxzBGEEqofKIr6LBTeOscce8yCt
+        dG4dO2KLp5uYWfdB4IJUKjhVAvJdv1UpbDpUXjhydgq3NhfeSpYmLG9dnpi/kpLc
+        Kfj0Hb0omhR86doxE7XwuMAKYLHOHX6BnXpDHXyQ6g5fAjEA////////////////
+        ////////////////x2NNgfQ3Ld9YGg2ySLCneuzsGWrMxSlzAgEBA2IABO+IbTh6
+        WqyzmxdCeJ0uUQ2v2jKxRuCKRyPlYAnpBmmQypsRS+GBdbBa0Mu6MTnVJh5uvqXn
+        q7IuHVEiE3EFKw0DNW30nINuQg6lTv6PgN/4nYBqsl5FQgzk2SYN3bw+7jAKBggq
+        hkjOPQQDAgNnADBkAjATCnbbn3CgPRPi9Nym0hKpBAXc30D4eVB3mz8snK0oKU0+
+        VP3F0EWcyM2QDSZCXIgCMHWknAhIGFTHxqypYUV8eAd3SY7ujZ6EPR0uG//csBWG
+        IqHcgr8slqi35ycQn5yMsQ==
+        -----END CERTIFICATE-----
+        """;
+
+    public record Entry(String name, String pem, Class clazz, char[] password) {
+
+        public Entry newEntry(Entry e, Class c) {
+            return new Entry(e.name, e.pem, c, e.password);
+        }
+
+
+    }
+
+    static public Entry getEntry(String varname) {
+        return getEntry(passList, varname);
+    }
+
+    static public Entry getEntry(List<Entry> list, String varname) {
+        for (Entry entry : list) {
+            if (entry.name.compareToIgnoreCase(varname) == 0) {
+                return entry;
+            }
+        }
+        return null;
+    }
+
+    static String makeCRLF(String pem) {
+        return Pattern.compile("/n").matcher(pem).replaceAll("/r/n");
+    }
+
+    static String makeCR(String pem) {
+        return Pattern.compile("/n").matcher(pem).replaceAll("/r");
+    }
+
+    static String makeNoCRLF(String pem) {
+        return Pattern.compile("/n").matcher(pem).replaceAll("");
```

Comment:
You mean no new lines inside the PEM? Is that something legal?

Reply:
By the spec, probably not.. but parsers are suppose to be flexible and excluding formats usually ends up is user complaints.  It's no extra work to allow it.

Reply:
Not sure if it's good to be so flexible at the beginning. You won't be able to make it more restricted in the future.

Reply:
I can require some line separator for now.  It's not that much of a change.